apis/network/v1alpha3/types.go

```diff
@@ -225,3 +225,249 @@ type SubnetList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []Subnet `json:"items"`
 }
+
+//Azure Firewall Structs
+// +kubebuilder:object:root=true
+// A AzureFirewall is a managed resource that represents an Azure Firewall
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="STATE",type="string",JSONPath=".status.state"
+// +kubebuilder:printcolumn:name="LOCATION",type="string",JSONPath=".spec.location"
+// +kubebuilder:printcolumn:name="RECLAIM-POLICY",type="string",JSONPath=".spec.reclaimPolicy"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories={crossplane,managed,azure}
+type AzureFirewall struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AzureFirewallSpec   `json:"spec"`
+	Status AzureFirewallStatus `json:"status,omitempty"`
+	///Properties SecurityGroupPropertiesFormat   `json:"properties,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// AzureFirewallList contains a list of Security Groups
+type AzureFirewallList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []AzureFirewall `json:"items"`
+}
+
+// A AzureFirewallSpec defines the desired state of a AzureFirewall.
+type AzureFirewallSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+
+	// ResourceGroupName - Name of the SecurityGroup's resource group.
+	ResourceGroupName string `json:"resourceGroupName,omitempty"`
+
+	// ResourceGroupNameRef - A reference to the the SecurityGroup's resource
+	// group.
+	ResourceGroupNameRef *runtimev1alpha1.Reference `json:"resourceGroupNameRef,omitempty"`
+
+	// ResourceGroupNameSelector - Select a reference to the the Azure Firewall
+	// resource group.
+	ResourceGroupNameSelector *runtimev1alpha1.Selector `json:"resourceGroupNameSelector,omitempty"`
+
+	// Location - Resource location.
+	Location string `json:"location"`
+
+	//AzureFirewallPropertiesFormat - Properties of AzureFirewall
+	AzureFirewallPropertiesFormat `json:"properties,omitempty"`
```

Comment:
If this is optional it should have `// +optional`

Reply:
'AzureFirewallPropertiesFormat' is not optional.